engineioxide/src/payload.rs

```diff
@@ -0,0 +1,164 @@
+use std::{io::BufRead, vec};
+use cfg_if::cfg_if;
+
+use crate::protocol::ProtocolVersion;
+
+pub const PACKET_SEPARATOR: u8 = b'\x1e';
+
+/// A payload is a series of encoded packets tied together.
+/// How packets are tied together depends on the protocol.
+pub struct Payload<R: BufRead> {
+    reader: R,
+    buffer: Vec<u8>,
+    #[allow(dead_code)]
+    protocol: ProtocolVersion,
+}
+
+type Item = Result<String, String>;
```

Comment:
You should create a custom Error type in the `errors` mod rather than using strings

Reply:
So the initial idea was to return a more generic error message here and catch it higher up for debugging only. But I'll make changes anyways because the messages ended up becoming quite similar.